feature/experimental/tunnel/otg_tests/tunnel_interface_based_ipv4_gre_encapsulation_test/tunnel_interface_based_ipv4_gre_encapsulation_test.go

```diff
@@ -0,0 +1,482 @@
+	// V4 destination
+	f1v4.Dst().Increment().SetStart(encapInnerDesIpv4Network).SetCount(200)
+	// Add L4 protocol
+	flow1ipv4.Packet().Add().Tcp()
+	// Increment Source port
+	flow1ipv4.Packet().Add().Tcp().SrcPort().Increment().SetStart(9000).SetCount(28000)
+	// Increment destination port
+	flow1ipv4.Packet().Add().Tcp().DstPort().Increment().SetStart(37001).SetCount(28000)
+
+}
+func fetchNetworkAddress(t *testing.T, address string, mask int) (string, string) {
+	addr := net.ParseIP(address)
+	var network net.IP
+
+	// This mask corresponds to a /24 subnet for IPv4.
+	ipv4Mask := net.CIDRMask(mask, 32)
+	network = addr.Mask(ipv4Mask)
+
+	networkWithMask := network.String() + "/" + strconv.Itoa(mask)
+	networkAlone := network.String()
+	//t.Logf("Network address : %s", networkWithMask)
+	return networkAlone, networkWithMask
+}
+func incrementAddress(t *testing.T, address string, i int, part string) string {
+	addr := net.ParseIP(address)
+	IsIPv4 := addr.To4()
+	var oct int
+	if IsIPv4 != nil {
+		if part == "network" {
+			oct = 2
+		} else if part == "host" {
+			oct = 3
+		}
+		for j := 0; j < i; j++ {
+			IsIPv4[oct]++
+		}
+	} else {
+		if part == "network" {
+			oct = 13
+		} else if part == "host" {
+			oct = 15
+		}
+
+		for j := 0; j < i; j++ {
+			addr[oct]++
+		}
+	}
+	return addr.String()
+}
+func configureTunnelBaseOnDUT(t *testing.T, dut *ondatra.DUTDevice, dp *ondatra.Port, a *attrs.Attributes) {
+	dutIntfs := []struct {
+		desc     string
+		intfName string
+		ipAddr   string
+		ipv4mask uint8
+		mac      string
+	}{
+		{
+			desc:     a.Desc,
+			intfName: dp.Name(),
+			ipAddr:   a.IPv4,
+			ipv4mask: a.IPv4Len,
+			mac:      a.MAC,
+		},
+	}
+	// Configure the interfaces.
+	for _, intf := range dutIntfs {
+		t.Logf("Configure DUT interface %s with attributes %v", intf.intfName, intf)
+		i := &oc.Interface{
+			Name:        ygot.String(intf.intfName),
+			Description: ygot.String(intf.desc),
+			Type:        oc.IETFInterfaces_InterfaceType_ethernetCsmacd,
+			Enabled:     ygot.Bool(true),
+		}
+		// configure ipv4 address
+		e := i.GetOrCreateEthernet()
+		e.MacAddress = ygot.String(intf.mac)
+		i4 := i.GetOrCreateSubinterface(0).GetOrCreateIpv4()
+		a := i4.GetOrCreateAddress(intf.ipAddr)
+		a.PrefixLength = ygot.Uint8(intf.ipv4mask)
+		gnmi.Replace(t, dut, gnmi.OC().Interface(intf.intfName).Config(), i)
+
+	}
+}
+func configureTunnelInterface(t *testing.T, intf string, unit int, tunnelSrc string, tunnelDst string, tunnelIpv4address string, Ipv4Mask int, dut *ondatra.DUTDevice) {
+	// IPv4 tunnel source and destination configuration
+	t.Logf("Push the IPv4 tunnel endpoint config:\n%s", dut.Vendor())
+	var config string
+	switch dut.Vendor() {
+	case ondatra.JUNIPER:
+		config = configureTunnelEndPoints(intf, unit, tunnelSrc, tunnelDst, tunnelIpv4address, Ipv4Mask)
+		t.Logf("Push the CLI config:\n%s", config)
+
+	default:
+		t.Errorf("Invalid Tunnel endpoint configuration")
```

Comment:
Make error msg more specific. Something like:

  _Tunnel endpoint configuration is not defined for dut.Vendor()._